### PR TITLE
Fix typos and remove trailing spaces

### PR DESCRIPTION
**Pull request by kianmeng**

### CODE REVIEW

**Review by mitya57 — APPROVED** [review]

File: docs/extensions/api.md [hunk 1]
```diff
@@ -852,7 +852,7 @@ assert someitem in registry
     * `priority`: An integer or float used to sort against all items.
 
     If an item is registered with a "name" which already exists, the existing item is replaced with the new item.
-    Tread carefully as the old item is lost with no way to recover it. The new item will be sorted according to its
+    Treat carefully as the old item is lost with no way to recover it. The new item will be sorted according to its
```
Comment (waylan) [inline docs/extensions/api.md]: This is not misspelled. "tread carefully" is an idiom which means 'proceed with caution.' If this is difficult to understand for non-native speakers, then I will consider proposals for rewording it. However "treat carefully" sounds awkward to my native ears. Either "be carefull" or "proceed with caution" could work.

**Comment by mitya57** [comment]
Can you please split typo fixes and removal of trailing whitespace into separate commits?

**Review by waylan — APPROVED** [review]
Thanks for the update. However, it is best to add another commit rather than force-pushing a  squashed commit. I appreciate that you are trying to provide a clean commit history. But you have obscured what changes were made before and after the previous review. Besides, Github gives us the ability to squash the commits when we merge the PR, so we will get a clean commit history anyway. No need to make any more changes here, but now you know for future reference.

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**